include/boost/iterator/new_iterator_tests.hpp

```diff
@@ -28,54 +28,45 @@
 // 04 Feb 2001  Added lvalue test, corrected preconditions
 //              (David Abrahams)
 
-# include <iterator>
-# include <boost/static_assert.hpp>
-# include <boost/concept_archetype.hpp> // for detail::dummy_constructor
-# include <boost/pending/iterator_tests.hpp>
-# include <boost/iterator/is_readable_iterator.hpp>
-# include <boost/iterator/is_lvalue_iterator.hpp>
-# include <boost/type_traits/is_same.hpp>
-# include <boost/mpl/bool.hpp>
-# include <boost/mpl/and.hpp>
+#include <boost/concept_archetype.hpp> // for detail::dummy_constructor
+#include <boost/iterator/is_lvalue_iterator.hpp>
+#include <boost/iterator/is_readable_iterator.hpp>
+#include <boost/pending/iterator_tests.hpp>
+#include <iterator>
+#include <type_traits>
 
-# include <boost/iterator/detail/config_def.hpp>
-# include <boost/detail/is_incrementable.hpp>
-# include <boost/core/lightweight_test.hpp>
+#include <boost/core/lightweight_test.hpp>
+#include <boost/detail/is_incrementable.hpp>
+#include <boost/iterator/detail/config_def.hpp>
+#include <boost/iterator/detail/type_traits/conjunction.hpp>
 
 namespace boost {
 
-
 // Do separate tests for *i++ so we can treat, e.g., smart pointers,
 // as readable and/or writable iterators.
 template <class Iterator, class T>
-void readable_iterator_traversal_test(Iterator i1, T v, mpl::true_)
-{
-    T v2(*i1++);
-    BOOST_TEST(v == v2);
+void readable_iterator_traversal_test(Iterator i1, T v, std::true_type) {
```

Comment:
Please, keep the braces on the next line. Here and everywhere else.

As a general comment, it is preferable to minimize code formatting changes for easier review. This PR is already quite large.